hal/src/stm32f2xx/bkpreg_hal.c

```diff
@@ -17,7 +17,7 @@ int32_t HAL_Core_Backup_Register(uint32_t BKP_DR)
 
 void HAL_Core_Write_Backup_Register(uint32_t BKP_DR, uint32_t Data)
 {
-    uint32_t BKP_DR_Address = HAL_Core_Backup_Register(BKP_DR);
+    int32_t BKP_DR_Address = HAL_Core_Backup_Register(BKP_DR);
```

Comment:
I guess the backup register addresses at this point are known to be less than `0x7FFFFFFF`  -- but it does seem weird that `HAL_Core_Backup_Register` returns an `int32_t`

Reply:
I think there is a bit of a confusion in these APIs and what is meant to be returned here is actually a backup register 'index' if you look at `RTC_BKP_DRX` definitions and also how it's implemented in Gen 3 (index in a lookup table), so it does make sense that `-1` would be 'unknown'.